migrate_fields_config.py

```diff
@@ -0,0 +1,222 @@
+    """
+    v1_field_names = v1_fields.keys()
+    v2_field_names = v2_fields.keys()
+    diffs = list()
+    for v1_field_name in v1_field_names:
+        if v1_field_name in v2_field_names:
+            v1_field = v1_fields[v1_field_name]
+            v2_field = v2_fields[v1_field_name]
+            diff = get_fields_difference(v1_field, v2_fields[v1_field_name])
+            if diff:
+                diffs.append((v1_field_name, {KEY_V1_FIELD: v1_field, KEY_V2_FIELD: v2_field, KEY_CONFIG_DIFF: diff}))
+    return diffs
+
+
+def v2_get_updated_field(field_difference):
+    """ Get a modified field according to the differences provided (CloudV2)
+    >>> diffs = ('field1', {'v1Field': {'sort': True, 'facet': True}, 'v2Field': {'name': 'field1', 'sort': False, 'facet': False, 'type': 'SOMETYPE'}, 'configDiff': [{'v1ConfigName': 'facet', 'v2ConfigName': 'facet', 'v1Value': True, 'v2Value': False}, {'v1ConfigName': 'sort', 'v2ConfigName': 'sort', 'v1Value': True, 'v2Value': False}]})
+    >>> v2_get_updated_field(diffs)
+    {'name': 'field1', 'sort': True, 'facet': True, 'type': 'SOMETYPE'}
+    """
+    for diff in field_difference[1][KEY_CONFIG_DIFF]:
+        # print(f'\t- "{diff[KEY_V2_CONFIGNAME]}" is "{diff[KEY_V2_VALUE]}" but should be "{diff[KEY_V1_VALUE]}"')
+        if field_difference[1][KEY_V2_FIELD]['type'].lower() == 'date' and \
+                diff[KEY_V2_CONFIGNAME].lower() == 'sort' and \
+                diff[KEY_V1_VALUE] == False:
+            print(f'\t-> Field "{field_difference[1][KEY_V2_FIELD]["name"]}" is of type date and cannot be set to sort = false. Skipping this change.')
+        else:
+            field_difference[1][KEY_V2_FIELD][diff[KEY_V2_CONFIGNAME]] = diff[KEY_V1_VALUE]
+    return field_difference[1][KEY_V2_FIELD]
+
+
+def v2_get_updated_fields(field_differences):
+    """ Get a list of modified fields according to the differences provided (CloudV2)
+    """
+    return [v2_get_updated_field(diff) for diff in field_differences]
+
+
+if __name__ == '__main__':
+    import doctest
+    if doctest.testmod().failed > 0:
+        exit(-1)
+
+    parser = argparse.ArgumentParser(description='Migrate fields configuration from CloudV1 to CloudV2')
+    parser.add_argument('--env', required=True, type=Environment, choices=list(Environment))
+    parser.add_argument('--v1_org_id', required=True)
+    parser.add_argument('--v1_source_name', required=True)
+    parser.add_argument('--v1_access_token', required=True)
+    parser.add_argument('--v2_org_id', required=True)
+    parser.add_argument('--v2_source_id', required=True)
+    parser.add_argument('--v2_access_token', required=True)
+    opts = parser.parse_args()
+
+    # args
+    env = opts.env
+    v1_org_id = opts.v1_org_id
+    v1_source_name = opts.v1_source_name
```

Comment:
Was there no source Id in V1?
I would use that instead of name if it's accessible (I don't remember)

Reply:
The source id is not that trivial to get in v1